.github/workflows/auto-merge.yml

```diff
@@ -9,8 +9,27 @@ permissions:
 jobs:
   dependabot:
     runs-on: ubuntu-latest
-    if: ${{ github.actor == 'dependabot[bot]' && contains(github.event.pull_request.labels.*.name, 'python') }}
+    if: ${{ github.actor == 'dependabot[bot]' }}
     steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1.1.1
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+
+      - name: Checkout branch
+        uses: actions/checkout@v2
+
+      - name: Check we're dealing with Python updates
+        run: |
+          if [[ $(git diff --name-only HEAD~1) == *"requirements"* ]]; then
```

Comment:
This seems a bit brittle to me.

I wonder if we can use `gh pr view --json labels -q .labels` ? gh cli is preinstalled, authed, and has the current PR context.

Reply:
Because we're fetching them via the api, the fact they are not in the event data shouldn't matter?

Reply:
Definitely up for giving it a go.  My assumption is dependabot applies them after the PR is created because it can't set them at PR creation time which is why we don't get them in the initial event payload, so there's some potential for a race condition.  At least this is checking the diff for expected changes!

Relatedly, I only expect us to use this sort of check for repos where we don't want all dependabot PRs merged, so eg the pipeline or databuilder repos wouldn't need this kind of check.

Reply:
Yeah, given the wait time for an action to run, I would be very surprised if the labels hadn't already been applied to the PR by the time this command runs